pagopa/pn-frontend
# PN-7440 - Caching payments response

## Short description
To avoid multiple requests to ext-registries we save the response on sessionStorage and use that to render payments when user refreshes or change page

## List of changes proposed in this pull request
- Set PaymentInfo response to sessionStorage
- On every request check if IUN is the same and 2 minutes haven't passed. If one of these two requirements is false, we recall ext-registries
- After a payment attempt when we return to SEND we request to ext-registries only the selected payment

## How to test
- When opening a notification detail should save iun and timestamp to sessionStorage and should require payments to ext-registries and cache them with pagination
- When refreshing page it shouldn't recall ext-registries but load payments from storage
- After a payment attempt it should call external registries with only one payment (the selected)
